lib/routes/account.js

```diff
@@ -333,6 +348,196 @@ module.exports = function (
     },
     {
       method: 'GET',
+      path: '/account/openid/authenticate',
+      handler: function (request, reply) {
+        var id = request.query.identifier
+        if (!isOpenIdProviderAllowed(id)) {
+          log.info({ op: 'Account.openid.authenticate', id: id })
+          return reply.redirect(
+            config.contentServer.url + '/openid?' + qs.stringify(
+              {
+                err: 'This OpenID Provider is not allowed'
+              }
+            )
+          )
+        }
+        openid.authenticate(
+          id,
+          config.openIdVerifyUrl,
+          null, // realm
+          false, // immediate
+          false, // stateless
+          function (err, authUrl) {
+            if (err) {
+              log.error({ op: 'Account.openid.authenticate', err: err })
+              return reply.redirect(
+                config.contentServer.url + '/openid?' + qs.stringify(
+                  {
+                    err: err.message
+                  }
+                )
+              )
+            }
+            reply.redirect(authUrl)
+          },
+          openIdExtensions,
+          false // strict
+        )
+      }
+    },
+    {
+      method: 'GET',
+      path: '/account/openid/login',
+      handler: function (request, reply) {
+        if (!request.url.search) {
```

Comment:
It would be helpful to have some high-level comments on the chunks of logic here, since the details of an OpenID flow are pretty niche.  Under what circumstances will this branch be triggered?  (And should we log something different to `Account.openid` when it is?)